dask/dataframe/groupby.py

```diff
@@ -293,6 +293,95 @@ def _var_agg(g, levels, ddof):
     return result
 
 
+def _cov_combine(g, levels):
+    return g
+
+
+def _cov_finalizer(df, cols):
+    vals = []
+    num_elements = len(list(it.product(cols, repeat=2)))
+    num_cols = len(cols)
+    vals = list(range(num_elements))
+    col_idx_mapping = dict(zip(cols, range(len(cols))))
+    for i, j in it.combinations_with_replacement(df[cols].columns, 2):
+        x = col_idx_mapping[i]
+        y = col_idx_mapping[j]
+        idx = x + num_cols * y
+        mul_col = "%s%s" % (i, j)
+        ni = df["%s-count" % i]
+        nj = df["%s-count" % j]
+
+        n = np.sqrt(ni * nj)
+        div = n - 1
+        div[div < 0] = 0
+        val = (df[mul_col] - df[i] * df[j] / n).values[0] / div.values[0]
+
+        vals[idx] = val
+        if i != j:
+            idx = num_cols * x + y
+            vals[idx] = val
+
+    level_1 = cols
+    index = pd.MultiIndex.from_product([level_1, level_1])
+    return pd.Series(vals, index=index)
+
+
+def mul_cols(df, cols):
+    _df = type(df)()
+    for i, j in it.combinations_with_replacement(df[cols].columns, 2):
+        col = "%s%s" % (i, j)
+        _df[col] = df[i] * df[j]
+    return _df
+
+
+def _cov_chunk(df, *index):
+    if is_series_like(df):
+        df = df.to_frame()
+    df = df.copy()
+    cols = df._get_numeric_data().columns
+
+    # when grouping by boolean series don't drop columns
+    is_mask = any(is_series_like(s) for s in index)
+    if not is_mask:
+        # casting to numpy may be problematic for non-pandas DataFrames
+        cols = cols.drop(np.array(index))
+    g = _groupby_raise_unaligned(df, by=index)
+    x = g.sum()
+
+    level = len(index)
+    mul = g.apply(mul_cols, cols=cols).reset_index(level=level, drop=True)
+    n = g[x.columns].count().rename(columns=lambda c: c + "-count")
```

Comment:
Perhaps use `.rename(columns"{}-count".format)`, to correctly handle integer column labels?